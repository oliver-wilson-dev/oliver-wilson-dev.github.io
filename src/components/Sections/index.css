.downloadButtonContainer {
    text-align: center;
    
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    margin: 40px 30px 0px 30px;
}