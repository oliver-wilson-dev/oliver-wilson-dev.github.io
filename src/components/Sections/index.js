import React from 'react';
import PropTypes from 'prop-types';
import Greeting from '../Greeting';
import Education from '../Education';
import Experience from '../Experience';
import LoadingSpinner from '../LoadingSpinner';
import Contact from '../Contact';


class Sections extends React.Component {
  componentDidMount() {
    const { props: { fetchContent } } = this;
    fetchContent();
  }

  render() {
    const { props: { sections, sectionsContentFetched } } = this;

    return (
      <React.Fragment>
        <LoadingSpinner loading={!sectionsContentFetched} />
        {sectionsContentFetched && (
        <React.Fragment>
          <Greeting {...sections.greeting} />
          <Education {...sections.education} />
          <Experience {...sections.experience} />
          <Contact {...sections.contact} />
        </React.Fragment>
        )}

      </React.Fragment>
    );
  }
}

Sections.propTypes = {
  fetchContent: PropTypes.func.isRequired,
  sections: PropTypes.object.isRequired,
  sectionsContentFetched: PropTypes.bool.isRequired
};

export default Sections;
