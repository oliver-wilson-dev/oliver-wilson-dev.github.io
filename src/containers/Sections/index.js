import { connect } from 'react-redux';
import Sections from '../../components/Sections';
import fetchContent from '../../state/actions/fetchContent';
import { getSections, getSectionsContentFetched } from '../../state/selectors';

const mapStateToProps = state => ({
  sections: getSections(state),
  sectionsContentFetched: getSectionsContentFetched(state)
});

const mapDispatchToProps = dispatch => ({
  fetchContent: () => dispatch(fetchContent())
});

export default connect(mapStateToProps, mapDispatchToProps)(Sections);
