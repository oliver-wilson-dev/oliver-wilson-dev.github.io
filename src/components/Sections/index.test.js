import React from 'react';
import { shallow } from 'enzyme';
import Sections from '.';

jest.mock('../Greeting', () => {
  const Greeting = () => null;

  return Greeting;
});

jest.mock('../Education', () => {
  const Education = () => null;

  return Education;
});

jest.mock('../Experience', () => {
  const Experience = () => null;

  return Experience;
});

jest.mock('../Contact', () => {
  const Contact = () => null;

  return Contact;
});

const defaultProps = {
  fetchContent: jest.fn(),
  sections: {
    greeting: { [Symbol('test-key')]: Symbol('test-value') },
    education: { [Symbol('test-key')]: Symbol('test-value') },
    experience: { [Symbol('test-key')]: Symbol('test-value') },
    contact: { [Symbol('test-key')]: Symbol('test-value') },
  },
  sectionsContentFetched: true
};

const render = overrideProps => shallow(<Sections {...defaultProps} {...overrideProps} />);

describe('<Sections/>', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });


  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should call fetchContent() once', () => {
    render();
    expect(defaultProps.fetchContent).toHaveBeenCalledWith();
  });

  describe('when the sections content is not fetched', () => {
    it('should render correctly', () => {
      expect(render({ sectionsContentFetched: false })).toMatchSnapshot();
    });
  });
});
