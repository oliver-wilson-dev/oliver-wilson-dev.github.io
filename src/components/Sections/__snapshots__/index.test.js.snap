// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sections/> should render correctly 1`] = `
<Fragment>
  <Greeting />
  <Education />
  <Experience />
  <Contact />
</Fragment>
`;

exports[`<Sections/> when the sections content is not fetched should render correctly 1`] = `<LoadingSpinner />`;
